### PR TITLE
Validate styles on block edit pages

### PR DESCRIPTION
**Pull request by mikeharv**
## Context
As part of the prep work to migrate Sprite Lab to Google Blockly, we will want to make sure all Sprite Lab blocks have style properties with valid values. If present, these styles will be used when rendering blocks instead of hard-coded HSV color values. This is critical to making our themes work as expected. The syntax for styles is also simpler, more maintainable, and easier to validate. We recently added styles to all the blocks that are defined in our levelbuilder block pools:
https://github.com/code-dot-org/code-dot-org/pull/50871 https://github.com/code-dot-org/code-dot-org/pull/50872 https://github.com/code-dot-org/code-dot-org/pull/50873 https://github.com/code-dot-org/code-dot-org/pull/50874
This data is not validated before being saved which means that blocks could potentially be created or modified without having a valid style listed. 

## Proposal
For block arguments with custom inputs, we validate the properties by comparing against a list of valid input types:
image
https://github.com/code-dot-org/code-dot-org/blob/mike/validate-styles-on-block-edit-pages/apps/src/block_utils.js#L1003-L1010

If we export a list of styles from our default theme palette, we can adapt the logic above to validate block styles.
Now, if a levelbuilder enters an unexpected style, they are alerted in realtime so that they can fix it before saving:

image

Styles are a new convention and the list shown above is admittedly pretty long and hard to parse. In anticipation of some common mistakes, we can use the first few letters entered to suggest a "best guess" of the style the levelbuilder was looking for:

image
image
image
If the first few letters aren't a good enough clue we can still provide the full list:
image

Going forwards, we expect a convention where all or most blocks have styles. Adding this validation takes away some of the guesswork. 

Like the best features, this change was immediately useful to me as it quickly revealed that I had actually shipped one invalid block styles in the PR linked in the top section:
image

(Fortunately, the `checkTouching` block is only usable today in CDO Blockly - which doesn't support these styles - so there's no chance that this bug impacted any user's experience.)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by molly-moen — COMMENTED** [review]
General questions: Does this only runs when editing a block? And it doesn't block the block from being saved if it is invalid? Basically I'm trying to figure out if this could cause any issues for blocks that already exist.

**Comment by mikeharv** [comment]
> General questions: Does this only runs when editing a block? And it doesn't block the block from being saved if it is invalid? Basically I'm trying to figure out if this could cause any issues for blocks that already exist.

@molly-moen It doesn't block the block from being saved. This also runs we install custom blocks, e.g. when we load a Sprite Lab project. We can effectively check that this won't cause issues for existing blocks by loading each block pool index and observing that there are no errors. I've done this on the branch, and it's how I caught the one block that wasn't configured correctly.

**Comment by mikeharv** [comment]
@molly-moen Also worth noting, @bencodeorg has been working on an "it renders" test that will likely check that all of the blocks in our custom pool indexes render as expected. If a levelbuilder did happen to change a block in a way that broke it, I think our test would hopefully pick up on that before the new levelbuilder content could go out.

**Review by molly-moen — APPROVED** [review]
